File: ru.kmorozov.library.data.server/src/main/java/ru/kmorozov/library/data/server/options/AbstractServerGBDOptions.java
package ru.kmorozov.library.data.server.options;

import ru.kmorozov.gbd.core.config.IGBDOptions;

public abstract class AbstractServerGBDOptions implements IGBDOptions {

    protected String bookId;

    @Override
    public String getBookId() {
        return bookId;
    }

    @Override
    public String getProxyListFile() {
        return null;
    }

    @Override
    public int getImageWidth() {
        return 0;
    }

    @Override
    public boolean reloadImages() {
        return false;
    }

    @Override
    public boolean secureMode() {
        return false;
    }

    @Override
    public String pdfOptions() {
        return null;
    }

    public void setBookId(String bookId) {
        this.bookId = bookId;
    }

}

File: ru.kmorozov.library.data.server/src/main/java/ru/kmorozov/library/data/server/options/AbstractServerGBDOptions.kt
package ru.kmorozov.library.data.server.options

import ru.kmorozov.gbd.core.config.IGBDOptions
import ru.kmorozov.gbd.core.config.IStorage
import ru.kmorozov.gbd.core.config.options.CtxOptions

abstract class AbstractServerGBDOptions : IGBDOptions {
    override var bookId: String = ""
        get() = TODO("not implemented") //To change initializer of created properties use File | Settings | File Templates.
    override val storage: IStorage
        get() = TODO("not implemented") //To change initializer of created properties use File | Settings | File Templates.
    override val proxyListFile: String
        get() = "tor"
    override val imageWidth: Int
        get() = 0

    override val reloadImages: Boolean
        get() = false

    override val secureMode: Boolean
        get() = true

    override fun pdfOptions(): String {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun ctxOptions(): CtxOptions {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

}
